demo/addons/fileimport/fileimport.js

```diff
@@ -0,0 +1,499 @@
+const fileimport = (function () {
+
+    var _button;
+    var _buttonActive = false;
+    var _importLayer;
+    var _toggleImportLayer = function() {
+        _buttonActive = !_buttonActive;
+        if (_importLayer) {
+            if (_buttonActive) {
+                mviewer.addLayer(_importLayer);
+            } else {
+                var el = $(`#layers-container [data-layerid=${_importLayer.layerid}]`)
+                mviewer.removeLayer(el);
+            }
+        } else {
+            var alertText = mviewer.lang
+            ? mviewer.lang[mviewer.lang.lang]("fileimport.alert.config")
+            : "Pas de couche d'import configurée"
+            mviewer.alert(alertText, "alert-warning");
+        }
+    }
+
+    var _wizard_modal =
```

Comment:
This file could be create separately and load with promise or simple ajax request.

1. Create template/fileimport.html  (contain this HTML code)
2. Add param into config file to register this file location 
3. In init, call from config url  and load this code to create modal

More clean but maybe not to fast.

Reply:
Sorry, but I also put it into the "nice to have" refacto

Reply:
> "nice to have"
I agree.